### PR TITLE
IO-Sim blog post

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

File: blog/2023-01-09-io-sim-annoucement.md [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+
+## IOSim on Hackage
+
+The IOG Networking Team is pleased to announce that we published [`io-sim`],
+[`io-classes`], [`si-timers`], [`strict-stm`] and [`io-classes-mtl`] on
+Hackage.  These are tools without which we could not imagine writing a complex
+distributed system like [Cardano].
+
+For developing a robust system one needs a proper testing framework which allows
+to model the key characteristics of the system.  The Cardano [network
```
Comment (njd42) [inline blog/2023-01-09-io-sim-annoucement.md:18]: To achieve this goal we need to create an abstraction that captures the key aspects of the Haskell runtime and operating system environment for distributed systems.

File: blog/2023-01-09-io-sim-annoucement.md [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+stack][ouroboros-network] is a highly concurrent system, and as a network
+application it needs to deal with time: there are all sorts of timeouts which
+guards resource usage: inactivity timeouts, message timeouts, or a application
+level `TCP`'s `WAIT_TIMEOUT` among others.
+
```
Comment (njd42) [inline blog/2023-01-09-io-sim-annoucement.md:23]: This permits us to capture issues related to timing (which abound in network programming) which, in production, would be extremely rare (things like simultaneous open) and ensure that we can test (in the simulation) those races.

File: blog/2023-01-09-io-sim-annoucement.md [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+[`io-classes-mtl`]).  Where we diverged from `base` (and other well
+established packages) we rather create a new package, hence `strict-stm` and
+`si-timers`.
+
+`io-classes` also support `MVar`'s and in the future it will also support
+`IOVar`s.  The `IO` instance of `MonadMVar` is using `GHC`'s `MVar`, while
+`IOSim` instance is based on a `TVar`.  `MVar`s, whether native of simulated,
+provide fairness but don't provide compositionality (like `TVar`s do via
+`STM`).
+
```
Comment (njd42) [inline blog/2023-01-09-io-sim-annoucement.md:39]: You might want to list (as bullets) all the 'iO' features that are captured here:

- STM
- Exceptions
- .... 

I realise that this is a blog post, but we need to draw in the reader - show them how comprehensive the abtraction is, shout out that _all_ the code bases that are using it (i.e show it has very wide applicability)

**Review by njd42 — COMMENTED** [review]
I added some points

File: blog/2023-01-09-io-sim-annoucement.md [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+---
+
+## IOSim on Hackage
+
+The IOG Networking Team is pleased to announce that we published [`io-sim`],
+[`io-classes`], [`si-timers`], [`strict-stm`] and [`io-classes-mtl`] on
+Hackage.  These are tools without which we could not imagine writing a complex
+distributed system like [Cardano].
+
```
[the remark this answers is not in the document]
Comment (coot) [reply blog/2023-01-09-io-sim-annoucement.md:16]: We can expand on this and include the `contra-tracer` into the picture.

**Pull request merged**